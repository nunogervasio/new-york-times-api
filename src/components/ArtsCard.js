import React from "react";
import { Grid, Header, Divider } from "semantic-ui-react";

const MainCard = ({ headline, summary, img_url, section }) => {
  return (
    <React.Fragment>
      <Grid>
        <Grid.Column width={6}>
          <Header as={"h3"}>
            <Header.Subheader>{section}</Header.Subheader>
            {headline}
          </Header>
        </Grid.Column>
        <Grid.Column width={10}>
          <p>{summary}</p>
          <p>{img_url}</p>
        </Grid.Column>
      </Grid>
      <Divider />
    </React.Fragment>
  );
};

export default MainCard;
