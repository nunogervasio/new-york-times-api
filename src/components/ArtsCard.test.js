import React from "react";
import { shallow } from "enzyme";
import ArtsCard from "./ArtsCard";

describe("ArtsCard", () => {
  const props = { headline: "", summary: "", img_url: "", section: "" };
  const artsCard = shallow(<ArtsCard {...props} />);

  it("renders properly", () => {
    expect(artsCard).toMatchSnapshot();
  });
});
