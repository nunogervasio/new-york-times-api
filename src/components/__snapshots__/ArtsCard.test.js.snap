// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtsCard renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainCard
    headline=""
    img_url=""
    section=""
    summary=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Grid>
          <GridColumn
            width={6}
          >
            <Header
              as="h3"
            >
              <HeaderSubheader>
                
              </HeaderSubheader>
              
            </Header>
          </GridColumn>
          <GridColumn
            width={10}
          >
            <p>
              
            </p>
            <p>
              
            </p>
          </GridColumn>
        </Grid>,
        <Divider />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn
              width={6}
            >
              <Header
                as="h3"
              >
                <HeaderSubheader>
                  
                </HeaderSubheader>
                
              </Header>
            </GridColumn>,
            <GridColumn
              width={10}
            >
              <p>
                
              </p>
              <p>
                
              </p>
            </GridColumn>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Header
                as="h3"
              >
                <HeaderSubheader>
                  
                </HeaderSubheader>
                
              </Header>,
              "width": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h3",
                "children": Array [
                  <HeaderSubheader>
                    
                  </HeaderSubheader>,
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": [Function],
                },
                "",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  
                </p>,
                <p>
                  
                </p>,
              ],
              "width": 10,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Grid>
            <GridColumn
              width={6}
            >
              <Header
                as="h3"
              >
                <HeaderSubheader>
                  
                </HeaderSubheader>
                
              </Header>
            </GridColumn>
            <GridColumn
              width={10}
            >
              <p>
                
              </p>
              <p>
                
              </p>
            </GridColumn>
          </Grid>,
          <Divider />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn
                width={6}
              >
                <Header
                  as="h3"
                >
                  <HeaderSubheader>
                    
                  </HeaderSubheader>
                  
                </Header>
              </GridColumn>,
              <GridColumn
                width={10}
              >
                <p>
                  
                </p>
                <p>
                  
                </p>
              </GridColumn>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Header
                  as="h3"
                >
                  <HeaderSubheader>
                    
                  </HeaderSubheader>
                  
                </Header>,
                "width": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h3",
                  "children": Array [
                    <HeaderSubheader>
                      
                    </HeaderSubheader>,
                    "",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": [Function],
                  },
                  "",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <p>
                    
                  </p>,
                  <p>
                    
                  </p>,
                ],
                "width": 10,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
